index.js

```diff
@@ -1,12 +1,15 @@
 var Buffer = require('safe-buffer').Buffer
 
 module.exports = function xor (a, b) {
-  var length = Math.max(a.length, b.length)
-  var buffer = Buffer.allocUnsafe(length)
+  var min = Math.min(a.length, b.length)
+  var max = Math.max(a.length, b.length)
+  var buffer = Buffer.allocUnsafe(max)
```

Comment:
if we did `var buffer = Buffer.from(<a or b whichever is longer>)`, we could just do an in place update of the longer one from the shorter or equal one.

Reply:
@calvinmetcalf wouldn't that mean iterating twice for large `Buffer`s?

Reply:
depends on how `Buffer.from(<a or b whichever is longer>)` ends up working behind the scene this might not be that much slower then allocUnsafe, so could be faster when we avoid an extra copy for different sized ones, so it would be worth testing.

Reply:
In case we execute our code on browser we wouldn't receive any benefit:
https://github.com/feross/buffer/blob/master/index.js#L241-L265
Buffer.from() will do the same, allocUnsafe() and copy().

If we execute our code NodeJs, it would be a worth testing.